### ISSUE TITLE
SSO session uses `/logout` endpoint when should be using `/disconnect/<backend>/` for proper SSO logout

### ISSUE THREAD

**Issue by whitej6** [nautobot/nautobot#4939]
### Environment
* Nautobot version (Docker tag too if applicable): All
* Python version: NA
* Database platform, version: NA
* Middleware(s): NA

### Steps to Reproduce
1. Signin with SSO
2. click logout button
3.

### Expected Behavior
Logout button should point to the disconnect endpoint that is namespaced to the backend which will inform the IDP of the user disconnecting their session in Nautobot.

### Observed Behavior
Logout button uses django logout functionality and does not inform the IDP os the disconnected session.

**Comment by gsnider2195** [comment]
Are we able to add the SSO logout functionality to the existing `/logout` view?

**Comment by whitej6** [comment]
No, you need to use the social auth core view. The easy patch could be to have a conditional in the template to render `/disconnect/<backend>/` when logged in as a social user. There's no additional views required to be developed or even enabled as [this URL path include](https://github.com/nautobot/nautobot/blob/develop/nautobot/core/urls.py#L51) publishes the endpoint today into nautobot.

TLDR - a one line change in the template can resolve this.

**Comment by gsnider2195** [comment]
Sorry I think I worded that wrong. I was wondering if we could update this so that requesting the `/logout` endpoint maybe redirects you to the `/disconnect/<backend>/` url?

**Comment by whitej6** [comment]
That should be possible

**Comment by glennmatthews** [comment]
I'm actually wondering if this is in fact a desirable behavior change. **Should** logging out of Nautobot actually log the user out of their SSO altogether? It feels like in an enterprise environment, using SSO all over the place, this would not necessarily be desirable.

**Comment by bryanculver** [comment]
Disconnect here seems more destructive than intended: https://python-social-auth.readthedocs.io/en/latest/logging_out.html

It appears it would fully remove the identity which would be unintended.
